src/csv/rowsReader/rowsReader.go

```diff
@@ -80,35 +84,37 @@ func (r *CsvReader) Err() error {
 	return r.scanner.Err()
 }
 
-// Search for \n -> rows delimiter. \n between enclosures is ignored.
-func splitRowsFunc(data []byte, atEOF bool) (advance int, token []byte, err error) {
-	length := len(data)
-
-	// Iterate over each character
-	insideEnclosure := false
-	for index, char := range data {
-		switch char {
-		case CsvLineBreak:
-			if !insideEnclosure {
-				// Line break outside enclosure -> row delimiter, return row
-				return index + 1, data[0 : index+1], nil
+func getSplitRowsFunc(enclosure byte) bufio.SplitFunc {
+	// Search for \n -> rows delimiter. \n between enclosures is ignored.
+	return func(data []byte, atEOF bool) (advance int, token []byte, err error) {
```

Comment:
Kedze sa `enclosure` konfiguruje, tak som to musel zabalit este do jednej funckcie.